esmvalcore/preprocessor/_regrid_iris_esmf_regrid.py

```diff
@@ -0,0 +1,208 @@
+    """Get a tuple with the horizontal dimensions of a cube."""
+
+    def _get_dims_along_axis(cube, axis):
+        try:
+            coord = cube.coord(axis=axis, dim_coords=True)
+        except iris.exceptions.CoordinateNotFoundError:
+            coord = cube.coord(axis=axis)
+        return cube.coord_dims(coord)
+
+    dims = {d for axis in ["x", "y"] for d in _get_dims_along_axis(cube, axis)}
+    return tuple(sorted(dims))
+
+
+class IrisESMFRegrid:
+    """:doc:`esmf_regrid:index` based regridding scheme.
+
+    Supports lazy regridding.
+
+    Parameters
+    ----------
+    method:
+        Either "conservative", "bilinear" or "nearest". Corresponds to the
+        :mod:`esmpy` methods
+        :attr:`~esmpy.api.constants.RegridMethod.CONSERVE`,
+        :attr:`~esmpy.api.constants.RegridMethod.BILINEAR` or
+        :attr:`~esmpy.api.constants.RegridMethod.NEAREST_STOD` used to
+        calculate regridding weights.
+    mdtol:
+        Tolerance of missing data. The value returned in each element of
+        the returned array will be masked if the fraction of masked data
+        exceeds ``mdtol``. ``mdtol=0`` means no missing data is tolerated while
+        ``mdtol=1`` will mean the resulting element will be masked if and only
+        if all the contributing elements of data are masked. If no value is
+        given, this will default to 1 for conservative regridding and 0
+        otherwise. Only available for methods 'bilinear' and 'conservative'.
+    use_src_mask:
+        If True, derive a mask from (first time step) of the source cube,
+        which will tell :mod:`esmpy` which points to ignore. If an array is
+        provided, that will be used.
+        If set to :obj:`None`, it will be set to :obj:`True` for methods
+        'bilinear' and `conservative' and to :obj:`False` for method 'nearest'.
+    use_tgt_mask:
+        If True, derive a mask from (first time step) of the target cube,
+        which will tell :mod:`esmpy` which points to ignore. If an array is
+        provided, that will be used.
+        If set to :obj:`None`, it will be set to :obj:`True` for methods
+        `bilinear' and 'conservative' and to :obj:`False` for method 'nearest'.
+    src_resolution:
+        If present, represents the amount of latitude slices per source cell
+        given to ESMF for calculation. If resolution is set, the source cube
+        must have strictly increasing bounds (bounds may be transposed
+        plus or minus 360 degrees to make the bounds strictly increasing).
+        Only available for method 'conservative'.
+    tgt_resolution:
+        If present, represents the amount of latitude slices per target cell
+        given to ESMF for calculation. If resolution is set, the target cube
+        must have strictly increasing bounds (bounds may be transposed
+        plus or minus 360 degrees to make the bounds strictly increasing).
+        Only available for method 'conservative'.
```

Comment:
I don't really understand this, but it looks like you just copied this from the regridder. In case you have a better understanding, could you rephrase it? For me, a "cell" contains one latitude, so multiple latitudes per cell sounds awkward...

Reply:
Yes, it's copy paste